pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -749,7 +748,7 @@ func groupPodsByServices(services []v1beta1.Service, pods v1beta1.GroupMemberPod
 	resolvedServices := make([]v1beta1.Service, len(services))
 	for podKey, pod := range pods {
 		for i := range services {
-			resolvedServices[i] = *resolveService(&services[i], pod)
+			resolvedServices[i] = *resolveService(&services[i], *pod.ToGroupMember())
```

Comment:
Switched to use pointer in resolveService. As for the comment about extra struct, do you think we need to make separate functions for groupMember and groupMemberPod?

Reply:
I used the benchmark test `BenchmarkGroupPodsByServicesWithNamedPort` to measure the difference.
```
# A separate resolveService for Pod
BenchmarkGroupPodsByServicesWithNamedPort-4           26          43131306 ns/op        13643059 B/op     401963 allocs/op
# Convert Pod to GroupMember first, then call resolveService
BenchmarkGroupPodsByServicesWithNamedPort-4           34          34344858 ns/op         8844193 B/op     301968 allocs/op
```
I think it's worth to avoid the conversion as this function is called quite frequently for named port case.

Reply:
Thanks for the benchmarking result. Updated.